Update Redis plans content (#196)
alphagov/paas-tech-docs: source/documentation/deploying_services/redis.md

> To set up a Redis service:

Reviewer: Possibly beyond the scope of this PR: There is a lot of repetition in rows 5-9, plus no text between the H2 and H3.

Author: Driven by accessibility header work. I needed to categorise the H3s.

1. Run the following in the command line to see what service plans are available for Redis:

    ```
    cf marketplace -s redis
    ```

    Here is an example of the output you will see (the exact service plans will vary):

    ```
    service plan            description                                                               free or paid
    tiny-3.2                568MB RAM, single node, no failover, daily backups                        free
    medium-ha-3.2           6.37GB RAM, highly-available, daily backups                               paid
    ```

>     Refer to the [Redis service plans](/deploying_services/redis/#redis-service-plans) section of the documentation for more information.

Reviewer: Could have the content from 'Redis service plans' here instead?

Author: Prefer to have the Redis service plans information in a separate section as it is supporting information.

> 

Reviewer: Beyond the scope of the PR: Why is data classification under 'Maintaining the service'?

Author: Agree

You can store data classified up to ‘official’ on the GOV.UK PaaS. Refer to the [data security classification documentation](/deploying_services/#data-security-classification) for more information.

> ### Redis service plans

Reviewer: Why is this section under 'Maintaining the service'?
I haven't found anywhere that says users have to deal with service plans regularly once they set up Redis...but maybe I've missed it?

There are two service plans currently available for the Redis service:

- `tiny-3.2`
- `medium-ha-3.2`

Do not use the `tiny-clustered-3.2` service plan for new service instances as we have deprecated this plan.

> Both service plans include encryption at rest of the database storage. This means that both the data on the disk and in snapshots is encrypted.

Author: the latter

> Amazon ElastiCache Redis backs up both service plans every day.

Author: agree

You can [vertically scale](/managing_apps.html#scaling) or upgrade your service plan.

#### Billing

If your org is in its trial period, you can use the `tiny-3.2` service plan for free.

Trial orgs cannot access paid service plans by default. If paid service plans are not enabled, when you try to use a paid service plan, you will receive an error stating “service instance cannot be created because paid service plans are not allowed”.

> One of your Org Managers must contact us at [gov-uk-paas-support@digital.cabinet-office.gov.uk](mailto:gov-uk-paas-support@digital.cabinet-office.gov.uk) to request that we enable paid service plans.

Reviewer: Does this apply to trial orgs or are there non-trial orgs that don't have paid service plans enabled?
If the latter applies, I think this information should be at the top of the section.

#### High availability

> You can use a high availability service plan (`HA`). When you use a high availability service plan, Amazon ElastiCache Redis provides a hot standby service for [failover](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/AutoFailover.html) [external link] in the event that the original service instance fails.

Reviewer: ```suggestion
If you use a high availability service plan, Amazon ElastiCache for Redis provides a hot standby service for [failover](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/AutoFailover.html) [external link].
```

The first sentence seemed redundant since this is a service plan option available to the user.

'failover' means transferring control to a duplicate system when a fault/failure happens so the last part seemed redundant.

Is there a need to introduce the `HA` abbreviation? And if so, why does it appear as inline code?

> If you have a non-HA service plan, you will lose data during a service instance failure. Before deciding which service plan to use, you should assess your data and what type of plan you need.

Reviewer: ```suggestion
If you don't have a high availability service plan, you will lose data during a service instance failure. Before deciding which service plan to use, you should assess your data and what type of plan you need.
```
To remove the need for abbreviation and (hopefully) make things clear in the absence of the abbreviation.

> If you have a non-HA service plan, you will lose data during a service instance failure. Before deciding which service plan to use, you should assess your data and what type of plan you need.

Author: agree

#### Billing

If your org is in its trial period, you can use the `tiny-3.2` service plan for free.

Trial orgs cannot access paid service plans by default. If paid service plans are not enabled, when you try to use a paid service plan, you will receive an error stating “service instance cannot be created because paid service plans are not allowed”.

One of your Org Managers must contact us at [gov-uk-paas-support@digital.cabinet-office.gov.uk](mailto:gov-uk-paas-support@digital.cabinet-office.gov.uk) to request that we enable paid service plans.

#### High availability

You can use a high availability service plan (`HA`). When you use a high availability service plan, Amazon ElastiCache Redis provides a hot standby service for failover in the event that the original service instance fails.

If you have a non-HA service plan, you will lose data during a service instance failure. Before deciding which service plan to use, you should assess your data and what type of plan you need.

Refer to the [Amazon ElastiCache documentation on failover](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/AutoFailover.html) [external link] for more information on failover.

> Refer to the [Redis documentation](https://redis.io/topics/cluster-tutorial) [external link] for more information on Redis overall.

Reviewer 2: I think we need to change this link to reflect the change in the text - this is still linking to the Redis clustering information. Do we even need this link at all?

Author: Removed and tweaked other link to be incorporated into the text
